### PR TITLE
Persona Pill Resources

### PR DESCRIPTION
**Pull request by jas29**
Solves issue #1293

### CODE REVIEW

File: app-web/src/utils/algolia.js [hunk 1]
```diff
@@ -101,19 +105,23 @@ export const reduceNodesForIndex = nodes => {
     } = node;
     let fields,
       content,
+      persona,
       id = node.id;
 
     if (type === 'GithubRaw') {
       fields = node.childMarkdownRemark.fields;
       content = node.childMarkdownRemark.content;
+      persona = node.childMarkdownRemark.frontmatter.personas;
```
Comment (patricksimonian) [inline app-web/src/utils/algolia.js]: isn't personas a field

File: app-web/src/utils/hooks.js [hunk 2]
```diff
@@ -52,15 +52,27 @@ export const useSearch = query => {
...
-          query: query,
-          hitsPerPage: 200, //set it to a large number, the default value is 200, which some search result is over that number
-        })
-        .then(res => {
-          setResults(res.hits);
-        });
+      if (query.includes('personas:')) {
+        let personaName = query.split(':')[1];
+        const filters = 'persona :' + personaName;
+        index
```
Comment (patricksimonian) [inline app-web/src/utils/hooks.js]: those two querieis could be refacgtored. only thing that changes is the search object

**Comment by sheaphillips** [comment]
Any chance you could expand scope on this item to deal with labels also? There’s a desperate ticket for it, but I think it will be at least as useful as fixing personas...

File: app-web/src/utils/algolia.js (hunk 1 above)
Comment (jas29) [reply app-web/src/utils/algolia.js]: it is a field for Devhub Siphon Nodes, however it exists within the frontmatter object for github raw nodes

**Comment by jas29** [comment]
> Any chance you could expand scope on this item to deal with labels also? There’s a desperate ticket for it, but I think it will be at least as useful as fixing personas...
> […](#)
> Sent from my iPhone
> On Mar 20, 2020, at 9:06 AM, Patrick Simonian ***@***.***> wrote: ﻿ @patricksimonian requested changes on this pull request. oops didn't mean to approve — You are receiving this because your review was requested. Reply to this email directly, view it on GitHub, or unsubscribe.

Yup Ill look into it :)

**Comment by patricksimonian** [comment]
@jas29 I'd do this in another pr. that work doesn't have to do with search so much as it has to do with creating dynamic topics based on that label. I think its an easy todo with a few challenges on labels. We shouldn't fire this off immediatly. Stay the course :)

**Comment by sheaphillips** [comment]
It's cool to do it as a separate PR.  I don't think we need dynamic topics,
though.  I *think* it's the same as the persona stuff - index the labels,
and then show clickable labels on pages/cards that run searches. (eg.
.../?query=label:cloud&label=azure).

File: app-web/src/utils/algolia.js (hunk 1 above)
Comment (patricksimonian) [reply app-web/src/utils/algolia.js]: lets make this a field for github raw nodes.

Comment (patricksimonian) [reply app-web/src/utils/algolia.js]: can be done in another pr

File: app-web/src/utils/hooks.js [hunk 3]
```diff
@@ -51,7 +51,15 @@ export const useSearch = query => {
   const [results, setResults] = useState([]);
   const index = searchClient.initIndex(`Devhub-Algolia-${ALGOLIA_INDEX_SUFFIX}`);
   useEffect(() => {
-    if (query) {
+    if (query && query.includes('persona:')) {
+      index
```
Comment (patricksimonian) [inline app-web/src/utils/hooks.js]: the search operation is the same. I think the code would look cleaner if we just did the search once and modified the arguments inside of it. 

```js

if foo
query = '';
options = {
  facetFilters: bar
}
else 
query = blah
options = {...}

// do search here

File: app-web/src/utils/hooks.js [hunk 4]
```diff
@@ -51,19 +51,20 @@ export const useSearch = query => {
...
-          setResults(res.hits);
-        });
+    let params;
+    let options = {};
+    if (query && query.includes('persona:')) {
+      params = '';
+      options.facetFilters = [query];
     } else {
-      setResults([]);
+      params = null;
```
Comment (patricksimonian) [inline app-web/src/utils/hooks.js]: seems like a bug?

File: app-web/src/utils/hooks.js [hunk 5]
```diff
@@ -51,16 +51,17 @@ export const useSearch = query => {
...
-          hitsPerPage: 200, //set it to a large number, the default value is 200, which some search result is over that number
-        })
-        .then(res => {
-          setResults(res.hits);
-        });
+      if (query && query.includes('persona:')) {
+        options.facetFilters = [query];
+      } else {
+        options.query = query;
+        options.hitsPerPage = 200;
```
Comment (patricksimonian) [inline app-web/src/utils/hooks.js]: 200 is the default hitsPerPage amount

**Pull request merged**